subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -214,13 +215,19 @@ public void execute(ExecutionListener executionListener) {
                 ParsedCommandLine parsedCommandLine = parser.parse(args);
 
                 buildLayoutConverter.convert(parsedCommandLine, buildLayout);
-                loggingConfigurationConverter.convert(parsedCommandLine, loggingConfiguration);
+
 
                 Map<String, String> properties = new HashMap<String, String>();
                 // Read *.properties files
                 layoutToPropertiesConverter.convert(buildLayout, properties);
                 // Read -D command line flags
                 systemPropertiesCommandLineConverter.convert(parsedCommandLine, properties);
+
+                // Convert properties for logging  object
+                PropertiesToLogLevelConfigurationConverter propertiesToLogLevelConfigurationConverter = new PropertiesToLogLevelConfigurationConverter();
+                propertiesToLogLevelConfigurationConverter.convert(properties, loggingConfiguration);
+                loggingConfigurationConverter.convert(parsedCommandLine, loggingConfiguration);
+
```

Comment:
I guess I don't see how that test is different than the test I added in CommandLineIntegrationSpec.groovy just above.